### ISSUE TITLE
droplet destroy --server results in ERROR: Warning all servers will be lost unless you exit with ctrl-c now!

### ISSUE THREAD

**Issue by hh** [rmoriz/knife-digital_ocean#76]
```
$ knife digital_ocean droplet destroy --server 5560345
ERROR: Warning all servers will be lost unless you exit with ctrl-c now!
^C^C^C^C^C^C!!!!
```

$ cat Gemfile
```
source 'https://rubygems.org'
gem 'chef-provisioning', github: 'chef/chef-provisioning'
gem 'chef-provisioning-fog', github: 'chef/chef-provisioning-fog'
gem 'berkshelf'
gem 'rubocop'
gem 'knife-digital_ocean'
gem "pry", "~> 0.10"
gem "pry-byebug"
gem "pry-stack_explorer"
gem 'thor'
```

Gemfile.lock
```
GIT
  remote: git://github.com/chef/chef-provisioning-fog.git
  revision: 7ea17f167c1ef28cc417f1665594835a3aade554
  specs:
    chef-provisioning-fog (0.13.2)
      chef
      chef-provisioning (~> 1.0)
      cheffish (>= 0.4)
      fog

GIT
  remote: git://github.com/chef/chef-provisioning.git
  revision: d7438a886a627d069a4ce4e9b59494aa7e93d94e
  specs:
    chef-provisioning (1.1.0)
      chef (>= 11.16.4)
      cheffish (~> 1.1)
      inifile (~> 2.0)
      net-scp (~> 1.0)
      net-ssh (~> 2.0)
      net-ssh-gateway (~> 1.2.0)
      winrm (~> 1.2.0)

GEM
  remote: https://rubygems.org/
  specs:
    CFPropertyList (2.3.1)
    activesupport (4.1.10)
      i18n (~> 0.6, >= 0.6.9)
      json (~> 1.7, >= 1.7.7)
      minitest (~> 5.1)
      thread_safe (~> 0.1)
      tzinfo (~> 1.1)
    addressable (2.3.8)
    akami (1.3.0)
      gyoku (>= 0.4.0)
      nokogiri
    ast (2.0.0)
    astrolabe (1.3.0)
      parser (>= 2.2.0.pre.3, < 3.0)
    axiom-types (0.1.1)
      descendants_tracker (~> 0.0.4)
      ice_nine (~> 0.11.0)
      thread_safe (~> 0.3, >= 0.3.1)
    berkshelf (3.2.3)
      addressable (~> 2.3.4)
      berkshelf-api-client (~> 1.2)
      buff-config (~> 1.0)
      buff-extensions (~> 1.0)
      buff-shell_out (~> 0.1)
      celluloid (~> 0.16.0)
      celluloid-io (~> 0.16.1)
      cleanroom (~> 1.0)
      faraday (~> 0.9.0)
      minitar (~> 0.5.4)
      octokit (~> 3.0)
      retryable (~> 2.0)
      ridley (~> 4.0)
      solve (~> 1.1)
      thor (~> 0.19)
    berkshelf-api-client (1.2.1)
      faraday (~> 0.9.0)
    binding_of_caller (0.7.2)
      debug_inspector (>= 0.0.1)
    buff-config (1.0.1)
      buff-extensions (~> 1.0)
      varia_model (~> 0.4)
    buff-extensions (1.0.0)
    buff-ignore (1.1.1)
    buff-ruby_engine (0.1.0)
    buff-shell_out (0.2.0)
      buff-ruby_engine (~> 0.1.0)
    builder (3.2.2)
    byebug (4.0.5)
      columnize (= 0.9.0)
    celluloid (0.16.0)
      timers (~> 4.0.0)
    celluloid-io (0.16.2)
      celluloid (>= 0.16.0)
      nio4r (>= 1.1.0)
    chef (12.2.1)
      chef-zero (~> 4.0)
      diff-lcs (~> 1.2, >= 1.2.4)
      erubis (~> 2.7)
      ffi-yajl (>= 1.2, < 3.0)
      highline (~> 1.6, >= 1.6.9)
      mixlib-authentication (~> 1.3)
      mixlib-cli (~> 1.4)
      mixlib-config (~> 2.0)
      mixlib-log (~> 1.3)
      mixlib-shellout (>= 2.0.0.rc.0, < 3.0)
      net-ssh (~> 2.6)
      net-ssh-multi (~> 1.1)
      ohai (~> 8.0)
      plist (~> 3.1.0)
      pry (~> 0.9)
      rspec-core (~> 3.2)
      rspec-expectations (~> 3.2)
      rspec-mocks (~> 3.2)
      rspec_junit_formatter (~> 0.2.0)
      serverspec (~> 2.7)
      specinfra (~> 2.10)
    chef-zero (4.2.1)
      ffi-yajl (>= 1.1, < 3.0)
      hashie (~> 2.0)
      mixlib-log (~> 1.3)
      rack
      uuidtools (~> 2.1)
    cheffish (1.1.2)
      chef (~> 12.1)
      chef-zero (~> 4.0)
    cleanroom (1.0.0)
    coderay (1.1.0)
    coercible (1.0.0)
      descendants_tracker (~> 0.0.1)
    columnize (0.9.0)
    debug_inspector (0.0.2)
    dep-selector-libgecode (1.0.2)
    dep_selector (1.0.3)
      dep-selector-libgecode (~> 1.0)
      ffi (~> 1.9)
    descendants_tracker (0.0.4)
      thread_safe (~> 0.3, >= 0.3.1)
    diff-lcs (1.2.5)
    droplet_kit (1.2.1)
      activesupport (~> 4.1.6)
      faraday (~> 0.9.1)
      kartograph (~> 0.2.1)
      resource_kit (~> 0.1.1)
      virtus (~> 1.0.3)
    equalizer (0.0.11)
    erubis (2.7.0)
    excon (0.45.2)
    faraday (0.9.1)
      multipart-post (>= 1.2, < 3)
    ffi (1.9.8)
    ffi-yajl (2.0.0)
      ffi (~> 1.5)
      libyajl2 (~> 1.2)
    fission (0.5.0)
      CFPropertyList (~> 2.2)
    fog (1.29.0)
      fog-atmos
      fog-aws (~> 0.0)
      fog-brightbox (~> 0.4)
      fog-core (~> 1.27, >= 1.27.4)
      fog-ecloud
      fog-json
      fog-local
      fog-powerdns (>= 0.1.1)
      fog-profitbricks
      fog-radosgw (>= 0.0.2)
      fog-riakcs
      fog-sakuracloud (>= 0.0.4)
      fog-serverlove
      fog-softlayer
      fog-storm_on_demand
      fog-terremark
      fog-vmfusion
      fog-voxel
      fog-xml (~> 0.1.1)
      ipaddress (~> 0.5)
      nokogiri (~> 1.5, >= 1.5.11)
    fog-atmos (0.1.0)
      fog-core
      fog-xml
    fog-aws (0.1.2)
      fog-core (~> 1.27)
      fog-json (~> 1.0)
      fog-xml (~> 0.1)
      ipaddress (~> 0.8)
    fog-brightbox (0.7.1)
      fog-core (~> 1.22)
      fog-json
      inflecto (~> 0.0.2)
    fog-core (1.30.0)
      builder
      excon (~> 0.45)
      formatador (~> 0.2)
      mime-types
      net-scp (~> 1.1)
      net-ssh (>= 2.1.3)
    fog-ecloud (0.1.1)
      fog-core
      fog-xml
    fog-json (1.0.1)
      fog-core (~> 1.0)
      multi_json (~> 1.0)
    fog-local (0.2.1)
      fog-core (~> 1.27)
    fog-powerdns (0.1.1)
      fog-core (~> 1.27)
      fog-json (~> 1.0)
      fog-xml (~> 0.1)
    fog-profitbricks (0.0.2)
      fog-core
      fog-xml
      nokogiri
    fog-radosgw (0.0.4)
      fog-core (>= 1.21.0)
      fog-json
      fog-xml (>= 0.0.1)
    fog-riakcs (0.1.0)
      fog-core
      fog-json
      fog-xml
    fog-sakuracloud (1.0.1)
      fog-core
      fog-json
    fog-serverlove (0.1.2)
      fog-core
      fog-json
    fog-softlayer (0.4.5)
      fog-core
      fog-json
    fog-storm_on_demand (0.1.1)
      fog-core
      fog-json
    fog-terremark (0.1.0)
      fog-core
      fog-xml
    fog-vmfusion (0.1.0)
      fission
      fog-core
    fog-voxel (0.1.0)
      fog-core
      fog-xml
    fog-xml (0.1.2)
      fog-core
      nokogiri (~> 1.5, >= 1.5.11)
    formatador (0.2.5)
    gssapi (1.0.3)
      ffi (>= 1.0.1)
    gyoku (1.3.1)
      builder (>= 2.1.2)
    hashie (2.1.2)
    highline (1.7.1)
    hitimes (1.2.2)
    httpclient (2.6.0.1)
    httpi (0.9.7)
      rack
    i18n (0.7.0)
    ice_nine (0.11.1)
    inflecto (0.0.2)
    inifile (2.0.2)
    ipaddress (0.8.0)
    json (1.8.2)
    kartograph (0.2.2)
    knife-digital_ocean (2.4.0)
      chef (>= 10.18)
      droplet_kit (~> 1.2.0)
    libyajl2 (1.2.0)
    little-plugger (1.1.3)
    logging (1.8.2)
      little-plugger (>= 1.1.3)
      multi_json (>= 1.8.4)
    method_source (0.8.2)
    mime-types (2.4.3)
    mini_portile (0.6.2)
    minitar (0.5.4)
    minitest (5.6.0)
    mixlib-authentication (1.3.0)
      mixlib-log
    mixlib-cli (1.5.0)
    mixlib-config (2.1.0)
    mixlib-log (1.6.0)
    mixlib-shellout (2.0.1)
    multi_json (1.11.0)
    multipart-post (2.0.0)
    net-http-persistent (2.9.4)
    net-scp (1.2.1)
      net-ssh (>= 2.6.5)
    net-ssh (2.9.2)
    net-ssh-gateway (1.2.0)
      net-ssh (>= 2.6.5)
    net-ssh-multi (1.2.1)
      net-ssh (>= 2.6.5)
      net-ssh-gateway (>= 1.2.0)
    nio4r (1.1.0)
    nokogiri (1.6.6.2)
      mini_portile (~> 0.6.0)
    nori (1.1.5)
    octokit (3.8.0)
      sawyer (~> 0.6.0, >= 0.5.3)
    ohai (8.2.0)
      ffi (~> 1.9)
      ffi-yajl (>= 1.1, < 3.0)
      ipaddress
      mime-types (~> 2.0)
      mixlib-cli
      mixlib-config (~> 2.0)
      mixlib-log
      mixlib-shellout (~> 2.0)
      rake (~> 10.1)
      systemu (~> 2.6.4)
      wmi-lite (~> 1.0)
    parser (2.2.0.pre.7)
      ast (>= 1.1, < 3.0)
      slop (~> 3.4, >= 3.4.5)
    plist (3.1.0)
    powerpack (0.0.9)
    pry (0.10.1)
      coderay (~> 1.1.0)
      method_source (~> 0.8.1)
      slop (~> 3.4)
    pry-byebug (3.1.0)
      byebug (~> 4.0)
      pry (~> 0.10)
    pry-stack_explorer (0.4.9.2)
      binding_of_caller (>= 0.7)
      pry (>= 0.9.11)
    rack (1.6.0)
    rainbow (2.0.0)
    rake (10.4.2)
    resource_kit (0.1.3)
      addressable (~> 2.3.6)
    retryable (2.0.1)
    ridley (4.1.2)
      addressable
      buff-config (~> 1.0)
      buff-extensions (~> 1.0)
      buff-ignore (~> 1.1)
      buff-shell_out (~> 0.1)
      celluloid (~> 0.16.0)
      celluloid-io (~> 0.16.1)
      erubis
      faraday (~> 0.9.0)
      hashie (>= 2.0.2, < 3.0.0)
      json (>= 1.7.7)
      mixlib-authentication (>= 1.3.0)
      net-http-persistent (>= 2.8)
      retryable (>= 2.0.0)
      semverse (~> 1.1)
      varia_model (~> 0.4)
    rspec (3.2.0)
      rspec-core (~> 3.2.0)
      rspec-expectations (~> 3.2.0)
      rspec-mocks (~> 3.2.0)
    rspec-core (3.2.3)
      rspec-support (~> 3.2.0)
    rspec-expectations (3.2.1)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.2.0)
    rspec-its (1.2.0)
      rspec-core (>= 3.0.0)
      rspec-expectations (>= 3.0.0)
    rspec-mocks (3.2.1)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.2.0)
    rspec-support (3.2.2)
    rspec_junit_formatter (0.2.0)
      builder (< 4)
      rspec (>= 2, < 4)
      rspec-core (!= 2.12.0)
    rubocop (0.27.0)
      astrolabe (~> 1.3)
      parser (>= 2.2.0.pre.6, < 3.0)
      powerpack (~> 0.0.6)
      rainbow (>= 1.99.1, < 3.0)
      ruby-progressbar (~> 1.4)
    ruby-progressbar (1.7.0)
    rubyntlm (0.1.1)
    savon (0.9.5)
      akami (~> 1.0)
      builder (>= 2.1.2)
      gyoku (>= 0.4.0)
      httpi (~> 0.9)
      nokogiri (>= 1.4.0)
      nori (~> 1.0)
      wasabi (~> 1.0)
    sawyer (0.6.0)
      addressable (~> 2.3.5)
      faraday (~> 0.8, < 0.10)
    semverse (1.2.1)
    serverspec (2.14.1)
      multi_json
      rspec (~> 3.0)
      rspec-its
      specinfra (~> 2.25)
    slop (3.6.0)
    solve (1.2.1)
      dep_selector (~> 1.0)
      semverse (~> 1.1)
    specinfra (2.28.3)
      net-scp
      net-ssh
    systemu (2.6.5)
    thor (0.19.1)
    thread_safe (0.3.5)
    timers (4.0.1)
      hitimes
    tzinfo (1.2.2)
      thread_safe (~> 0.1)
    uuidtools (2.1.5)
    varia_model (0.4.0)
      buff-extensions (~> 1.0)
      hashie (>= 2.0.2, < 3.0.0)
    virtus (1.0.5)
      axiom-types (~> 0.1)
      coercible (~> 1.0)
      descendants_tracker (~> 0.0, >= 0.0.3)
      equalizer (~> 0.0, >= 0.0.9)
    wasabi (1.0.0)
      nokogiri (>= 1.4.0)
    winrm (1.2.0)
      gssapi (~> 1.0.0)
      httpclient (~> 2.2, >= 2.2.0.2)
      logging (~> 1.6, >= 1.6.1)
      nokogiri (~> 1.5)
      rubyntlm (~> 0.1.1)
      savon (= 0.9.5)
      uuidtools (~> 2.1.2)
    wmi-lite (1.0.0)

PLATFORMS
  ruby

DEPENDENCIES
  berkshelf
  chef-provisioning!
  chef-provisioning-fog!
  knife-digital_ocean
  pry (~> 0.10)
  pry-byebug
  pry-stack_explorer
  rubocop
  thor

BUNDLED WITH
   1.10.2
```

**Comment by rmoriz** [comment]
What's exactly your problem? Knife doesn't exit with ctrl+c?

The used code is quite simple
https://github.com/rmoriz/knife-digital_ocean/blob/master/lib/chef/knife/digital_ocean_droplet_destroy.rb#L44-L47

**Comment by hh** [comment]
I am unable to delete any nodes at all.

```knife digital_ocean droplet list``` shows servers, and the ```droplet destroy`` command requires -S or --server to specify a droplet.

``` knife digital_ocean droplet destroy --server 5560345``` should destroy/delete the droplet, however without ```--all``` I get this message warning me that all droplets will be destroyed. I can hit ctrl-c, or specify --all but neither would result in destroying only the requested node.

**Comment by mateuszkwiatkowski** [comment]
Hello,
We're affected with this issue too. 'droplet destroy' command doesn't remove specified droplet.

**Comment by rmoriz** [comment]
Commit in https://github.com/rmoriz/knife-digital_ocean/commit/0a5acf4f1182d665e965b017f83502dbd926fcd4 was probably a mistake. Despite the non-ideal message, it seems to wait 15x15s = 3,75 minutes

**Comment by rmoriz** [comment]
I currently don't have a Digital Ocean test account to verify the change and run all the tests. Any volunteers? :-)

**Comment by rmoriz** [comment]
released 2.4.1
